src/cli/compiler.js

```diff
@@ -66,6 +67,80 @@ class Compiler {
     }
   }
 
+  getFileLocations() {
```

Comment:
I'd make the purpose of the function even more clear: `getFilesToWatch`, then rename `allFiles` to `files`.

Then I'd also drop the `let allAbsolutePaths = ` and simply return the result of `files.map(...)`.